Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -860,9 +854,25 @@ extension MapboxMap {
 
 extension MapboxMap: MapEventsObservable {
 
+    /// Listen to a single occurrence of a Map event.
+    ///
+    /// This will observe the next (and only the next) event of the specified
+    /// type. After observation, the underlying subscriber will unsubscribe from
+    /// the map or snapshotter.
+    ///
+    /// If you need to unsubscribe before the event fires, call `cancel()` on
+    /// the returned `Cancelable` object.
+    ///
+    /// - Parameters:
+    ///   - eventType: The event type to listen to.
+    ///   - handler: The closure to execute when the event occurs.
+    ///
+    /// - Returns: A `Cancelable` object that you can use to stop listening for
+    ///     the event. This is especially important if you have a retain cycle in
+    ///     the handler.
     @discardableResult
-    private func onNext(eventTypes: [MapEvents.EventKind], handler: @escaping (Event) -> Void) -> Cancelable {
-        return observable.onNext(eventTypes, handler: handler)
+    public func onTypedNext<Payload: Decodable>(_ eventType: MapEvents.Event<Payload>, handler: @escaping (TypedEvent<Payload>) -> Void) -> Cancelable {
+        return observable.onTypedNext(eventType, handler: handler)
```

Comment:
Can we discuss the API design here?
I think the API would be better if we preserve the same `onNext` prefix. As it might be type-checker issues, we can make it `onNext<…>(event: MapsEvents.Event<…>)`
What do you think?

Reply:
Nice idea, I renamed `onTypedNext/onTypedEvery` to `onNext(event:.../onEvery(event:...`